### PR TITLE
Safari in browser auth

### PR DESCRIPTION
**Pull request by kirlat**
Adds support for authentication in Safari using Auth0 SPA SDK. This build luckily has all the right terser versions so it builds and runs without any issues 🙂

### CODE REVIEW

File: src/content/content-safari.js [hunk 1]
```diff
@@ -208,11 +214,102 @@ const handleLogoutRequest = async function handleLogoutRequest (message) {
...
+    const searchRes = window.document.URL.match(/auth=(\d+)/)
+    if (!searchRes) {
+      // Do not inject anything if `auth` is not present
+      return
+    }
+    const authStatus = (searchRes && searchRes.length === 2 && searchRes[1] === '1')
+    let authDiv = document.createElement('div') // eslint-disable-line prefer-const
+
+    authDiv.innerHTML += authStatus
+      ? '<button id="alpheios-logout" class="alpheios-button-primary">Log Out</button>'
```
Comment (balmas) [inline src/content/content-safari.js]: we should retrieve these strings from the messages file

**Comment by kirlat** [comment]
> One question though -- should we now have a sessionRequestHandler?

Do you mean the one in the `background-process.js`? It is used by webextension where we employ `auth0-chrome`. Because we're not touching that workflow for now then probably that one should stay (if we're talking about the same thing).

File: src/content/content-safari.js (hunk 1 above)
Comment (kirlat) [reply src/content/content-safari.js]: Agree with that. We can probably do that as the next iteration (i.e. I can do another PR with those changes).

I was thinking about moving the whole code part that renders the UI in the iframe to the components. Maybe event creating a Vue UI component (effectively just a button) that will be pulled out from components by webextension and then mounted within an iframe. But maybe that'll be to much. What do you think?

Comment (balmas) [reply src/content/content-safari.js]: yes, probably a good idea. But let's wait until after we release this.

**Comment by balmas** [comment]
> > One question though -- should we now have a sessionRequestHandler?
> 
> Do you mean the one in the `background-process.js`? It is used by webextension where we employ `auth0-chrome`. Because we're not touching that workflow for now then probably that one should stay (if we're talking about the same thing).

`uiController.activate()` calls `auth.session()` to see if the user has an active session. In FF and Chrome, this results in a call to the auth0 userinfo endpoint to see if the token is still active. I am a little lost as to how we are handling this in Safari (and why we are doing it differently there now that we are doing client side auth)

**Pull request merged**